src/Symfony/Component/Cache/SimpleCache.php

```diff
@@ -0,0 +1,164 @@
+
+    public function __construct(CacheItemPoolInterface $pool)
+    {
+        $this->pool = $pool;
+
+        if ($pool instanceof Adapter\AdapterInterface) {
+            $this->createCacheItem = \Closure::bind(
+                function ($key, $value) {
+                    CacheItem::validateKey($key);
+                    $item = new CacheItem();
+                    $item->key = $key;
+                    $item->value = $value;
+
+                    return $item;
+                },
+                null,
+                CacheItem::class
+            );
+        } else {
+            $this->createCacheItem = function ($key, $value) {
+                return $this->pool->getItem($key)->set($value);
+            };
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($key, $default = null)
+    {
+        $item = $this->pool->getItem($key);
+
+        return $item->isHit() ? $item->get() : $default;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function set($key, $value, $ttl = null)
+    {
+        $f = $this->createCacheItem;
+        $item = $f($key, $value);
+        if (null !== $ttl) {
+            $item->expiresAfter($ttl);
+        }
+
+        return $this->pool->save($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete($key)
+    {
+        return $this->pool->deleteItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        return $this->pool->clear();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMultiple($keys)
+    {
+        foreach ($this->pool->getItems($keys) as $key => $item) {
+            yield $key => $item->get();
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setMultiple($items, $ttl = null)
+    {
+        $f = $this->createCacheItem;
+        $ok = true;
+
+        foreach ($items as $key => $value) {
+            $item = $f($key, $value);
+            if (null !== $ttl) {
+                $item->expiresAfter($ttl);
+            }
+            $ok = $this->pool->saveDeferred($item) && $ok;
+        }
+
+        return $this->pool->commit() && $ok;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteMultiple($keys)
+    {
+        return $this->pool->deleteItems($keys);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function has($key)
+    {
+        return $this->pool->hasItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function increment($key, $step = 1)
+    {
+        if ($this->pool instanceof CounterInterface) {
+            return $this->pool->increment($key, $step);
+        }
+        if (!is_numeric($step)) {
+            return false;
+        }
+        $step = (int) $step;
+
+        $item = $this->pool->getItem($key);
+        if (is_numeric($value = $item->get())) {
+            $step += $value;
+        }
+        $item->set($step);
```

Comment:
there is only the phpdoc here in the spec, and it says ``Increment atomically`` so it looks like a MUST for now (at least I understand it this way when reading it)